### ISSUE TITLE
Ansible user module fails when parent directory of home dir does not exist even when create_home=no

### ISSUE THREAD

**Issue by thanasis-liapis** [ansible/ansible#76733]
### Summary

Executing the user module to create a new user, when parent directory of new user's home dir does not exist, even when I choose not to create the home directory (create_home=no), the task fails with error

`fatal: [cmk2]: FAILED! => {"changed": false, "msg": "[Errno 2] No such file or directory: /root/dir/site3'"}`

The path `/root/dir` does not exist in the current phase and cannot be created (it is located in a shared disk currently mounted in another host). However, I choose not to create the home directory with `create_home=no`, so I would expect this to work.

Fyi, running the `useradd` command directly to the machine, with the same settings as the ones used in the user module, works as expected.

### Issue Type

Bug Report

### Component Name

user module

### Ansible Version

```console
$ ansible --version
ansible 2.9.7
  config file = ./ansible.cfg
  configured module search path = [u'~/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.17 (default, Feb 27 2021, 15:10:58) [GCC 7.5.0]
```

### Configuration

```console
$ ansible-config dump --only-changed
ANSIBLE_FORCE_COLOR(~/ansible.cfg) = True
ANSIBLE_PIPELINING(~/ansible.cfg) = True
ANSIBLE_SSH_ARGS(env: ANSIBLE_SSH_ARGS) = -o ForwardAgent=yes -o ServerAliveInterval=30 -o ControlMaster=auto -o ControlPersist=600s -o StrictHostKeyChecking=no -i vault/id_rsa -o IdentitiesOnly=yes
CACHE_PLUGIN_TIMEOUT(~/ansible.cfg) = 1
DEFAULT_CALLBACK_WHITELIST(~/ansible.cfg) = [u'profile_tasks', u'timer']
DEFAULT_FACT_PATH(~/ansible.cfg) = /etc/ansible/facts.d
DEFAULT_FORKS(~/ansible.cfg) = 1
DEFAULT_GATHERING(~/ansible.cfg) = smart
DEFAULT_GATHER_SUBSET(~/ansible.cfg) = [u'all']
DEFAULT_GATHER_TIMEOUT(~/ansible.cfg) = 1
DEFAULT_REMOTE_USER(env: ANSIBLE_REMOTE_USER) = ubuntu
DEFAULT_ROLES_PATH(~/ansible.cfg) = 
DEFAULT_VAULT_PASSWORD_FILE(~/ansible.cfg) = 
HOST_KEY_CHECKING(~/ansible.cfg) = True
USE_PERSISTENT_CONNECTIONS(~/ansible.cfg) = False
```

### OS / Environment

Target Host OS: Ubuntu 20.04.3 LTS
Target Host Python: Python 3.8.10

### Steps to Reproduce

```yaml (paste below)
- name: "Create operating system account [{{ item.name }}]"
  become: yes
  user:
    name: "{{ item.name }}"
    group: "{{ item.name }}"
    groups: "{{ mygroups }}"
    comment: "User {{ item.name }}"
    create_home: no
    home: "{{ root_dir }}/{{ item.name }}"
    password: 'xxxxxx'
    shell: "{{ login_shell }}"
    system: no
    state: present
    uid: "{{ item.uid }}"
  when: getent_passwd[item.name] == none
  no_log: true
```

### Expected Results

I expect the user to be created even if the parent directory of the home directory does not exist, when I do not choose to create the home directory `[create_home=no]`, similarly to the native `useradd` OS command.

### Actual Results

```console
Using module file /usr/local/lib/python2.7/dist-packages/ansible/modules/system/user.py
Pipelining is enabled.
<IP> ESTABLISH SSH CONNECTION FOR USER: ubuntu
<IP> SSH: EXEC ssh -vvv -o ForwardAgent=yes -o ServerAliveInterval=30 -o ControlMaster=auto -o ControlPersist=600s -o StrictHostKeyChecking=no -i vault/id_rsa -o IdentitiesOnly=yes -o Port=22 -o 'IdentityFile="./vault/id_rsa"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="ubuntu"' -o ConnectTimeout=10 -o ControlPath=~/.ansible/cp/10491fd927 IP '/bin/sh -c '"'"'sudo -H -S -n  -u root /bin/sh -c '"'"'"'"'"'"'"'"'echo BECOME-SUCCESS-nnmfrtnxnbflzhaxtobtirizzfufkkpd ; /usr/bin/python3'"'"'"'"'"'"'"'"' && sleep 0'"'"''
Escalation succeeded
<IP> (0, '\n{"failed": true, "msg": "[Errno 2] No such file or directory: \'/root/dir/site3\'", "invocation": {"module_args": {"comment": "User site3", "uid": 50003, "create_home": false, "home": "/root/dir/site3", "group": "site3", "system": false, "state": "present", "shell": "/bin/bash", "groups": ["omd"], "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", "name": "site3", "non_unique": false, "force": false, "remove": false, "move_home": false, "append": false, "ssh_key_bits": 0, "ssh_key_type": "rsa", "ssh_key_comment": "ansible-generated on cmk2", "update_password": "always", "login_class": null, "hidden": null, "seuser": null, "skeleton": null, "generate_ssh_key": null, "ssh_key_file": null, "ssh_key_passphrase": null, "expires": null, "password_lock": null, "local": null, "profile": null, "authorization": null, "role": null}}, "warnings": ["The input password appears not to have been hashed. The \'password\' argument must be encrypted for this module to work properly."]}\n', 'Warning: Identity file vault/id_rsa not accessible: No such file or directory.\nOpenSSH_7.6p1 Ubuntu-4ubuntu0.5, OpenSSL 1.0.2n  7 Dec 2017\r\ndebug1: Reading configuration data ~/.ssh/config\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 19: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 2 remote\r\ndebug1: Requesting forwarding of remote forward LOCALHOST:8022 -> MYHOST:22\r\ndebug1: Requesting forwarding of remote forward LOCALHOST:8080 -> MYPROXY:3128\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 20055\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
[WARNING]: The input password appears not to have been hashed. The 'password' argument must be encrypted for this module to work properly.
fatal: [cmk2]: FAILED! => {
    "changed": false, 
    "invocation": {
        "module_args": {
            "append": false, 
            "authorization": null, 
            "comment": "User site3", 
            "create_home": false, 
            "expires": null, 
            "force": false, 
            "generate_ssh_key": null, 
            "group": "site3", 
            "groups": [
                "omd"
            ], 
            "hidden": null, 
            "home": "/root/dir/site3", 
            "local": null, 
            "login_class": null, 
            "move_home": false, 
            "name": "site3", 
            "non_unique": false, 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "password_lock": null, 
            "profile": null, 
            "remove": false, 
            "role": null, 
            "seuser": null, 
            "shell": "/bin/bash", 
            "skeleton": null, 
            "ssh_key_bits": 0, 
            "ssh_key_comment": "ansible-generated on cmk2", 
            "ssh_key_file": null, 
            "ssh_key_passphrase": null, 
            "ssh_key_type": "rsa", 
            "state": "present", 
            "system": false, 
            "uid": 50003, 
            "update_password": "always"
        }
    }, 
    "msg": "[Errno 2] No such file or directory: '/root/dir/site3'"
}
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by sivel** [comment]
This appears to have been fixed in https://github.com/ansible/ansible/pull/70600 and included in 2.9.12

**Comment by thanasis-liapis** [comment]
I can confirm that after upgrading ansible the issue is resolved. The `user` module is no longer failing when the parent directory of the home directory does not exist and when create_home=no.

For the accuracy of my information, I upgraded to the, currently, latest of the 2.9 version - 2.9.27.

Thanks @sivel for your help.
